In [182]:
import pandas as pd
import matplotlib.pyplot as plt
df = pd.read_csv('data/raw/shopping.csv')
df.head()
df.info()
df.describe()


<class 'pandas.DataFrame'>
RangeIndex: 3900 entries, 0 to 3899
Data columns (total 15 columns):
 #   Column                  Non-Null Count  Dtype  
---  ------                  --------------  -----  
 0   Customer ID             3900 non-null   int64  
 1   Age                     3510 non-null   float64
 2   Gender                  3900 non-null   str    
 3   Item Purchased          3900 non-null   str    
 4   Purchase Amount (USD)   3900 non-null   float64
 5   Location                3510 non-null   str    
 6   Size                    3900 non-null   str    
 7   Color                   3900 non-null   str    
 8   Season                  3900 non-null   str    
 9   Review Rating           1431 non-null   float64
 10  Shipping Type           3900 non-null   str    
 11  Promo Code Used         3900 non-null   str    
 12  Previous Purchases      3900 non-null   int64  
 13  Payment Method          3900 non-null   str    
 14  Frequency of Purchases  1560 non-null   str    
dty

,Customer ID,Age,Purchase Amount (USD),Review Rating,Previous Purchases
count,3900.000000,3510.000000,3900.000000,1431.000000,3900.000000
mean,1950.500000,34.256695,38.708329,3.904962,5.841538
std,1125.977353,10.616340,11.504923,1.105952,5.843695
min,1.000000,18.000000,11.392520,1.000000,0.000000
25%,975.750000,26.000000,29.332617,4.000000,1.000000
50%,1950.500000,33.000000,36.953004,4.000000,4.000000
75%,2925.250000,41.000000,48.739596,5.000000,9.000000
max,3900.000000,75.000000,69.567481,5.000000,20.000000


In [183]:
df

,Customer ID,Age,Gender,Item Purchased,Purchase Amount (USD),Location,Size,Color,Season,Review Rating,Shipping Type,Promo Code Used,Previous Purchases,Payment Method,Frequency of Purchases
0,3475,NaN,Male,Jacket,30.904467,Maine,M,Burnt orange,Fall,4.0,Standard,No,0,Credit Card,NaN
1,3698,21.0,Female,Backpack,31.588259,NaN,L,Turquoise,Winter,2.0,Express,No,1,Credit Card,Monthly
2,2756,31.0,Male,Leggings,24.231704,Nevada,M,Terra cotta,Winter,4.0,Standard,No,0,Credit Card,NaN
3,3340,NaN,Male,Pajamas,33.918834,Nebraska,M,Black,Winter,NaN,Standard,No,2,Credit Card,Bi-Weekly
4,3391,38.0,Male,Sunglasses,36.545487,Oregon,S,Aubergine,Summer,NaN,Standard,No,0,Credit Card,NaN
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
3895,147,43.0,Female,Sunglasses,61.610602,Colorado,S,Burnt orange,Fall,NaN,2-Day Shipping,Yes,7,Venmo,NaN
3896,1472,37.0,Male,Pajamas,44.600556,Alaska,S,Aubergine,Winter,4.0,2-Day Shipping,Yes,20,Venmo,NaN
3897,1069,NaN,Female,Handbag,41.781965,Wyoming,M,Fuchsia,Summer,NaN,Standard,Yes,13,Venmo,NaN
3898,1171,39.0,Male,Hoodie,45.343778,Illinois,S,Brick red,Fall,NaN,2-Day Shipping,Yes,3,Venmo,NaN


In [184]:
# 1. Which clothing items are purchased most frequently by gender?
df.pivot_table(
    index='Item Purchased',
    columns='Gender',
    aggfunc='size',
    fill_value=0
).sort_values(by=['Female','Male'], ascending=False).head(10)


Gender,Female,Male,Non-binary
Item Purchased,,,
Shorts,159,152,3
Running Shoes,157,148,4
Sunglasses,149,161,2
T-shirt,143,119,0
Dress,142,151,1
Handbag,124,140,3
Socks,113,96,2
Pajamas,113,94,5
Backpack,112,110,3


The shorts is the clothing item that was purchased the most frequently. Female customers had the highest amount of shorts purchased compared to males. They purchased 7 more than males in this dataset.

In [185]:
# 2. How does purchasing frequency relate to whether customers leave a review and to the rating they give?
df['Left Review'] = df['Review Rating'].notnull()
df.groupby('Left Review')['Review Rating'].value_counts()

Left Review  Review Rating
True         4.0              715
             5.0              448
             2.0              218
             1.0               49
             3.0                1
Name: count, dtype: int64

In [186]:
df.groupby(['Left Review', 'Review Rating'])['Frequency of Purchases'].value_counts()

Left Review  Review Rating  Frequency of Purchases
True         1.0            Quarterly                   8
                            Monthly                     7
                            Bi-Weekly                   5
                            Annually                    1
             2.0            Quarterly                  38
                            Annually                   23
                            Monthly                    21
                            Bi-Weekly                   7
                            Weekly                      2
             4.0            Quarterly                 133
                            Monthly                    77
                            Annually                   58
                            Bi-Weekly                  31
                            Weekly                      9
             5.0            Quarterly                  81
                            Monthly                    45
                     

Most customers who left a review gave a 4‑star rating, making it the most common score with 715 total responses. Among those 4‑star reviewers, the largest group consisted of customers who typically make purchases on a quarterly basis, accounting for 133 customers.

In [187]:
# 3. Does the seasonality of an item influence the purchase amount customers spend?
df.groupby('Season')['Purchase Amount (USD)'].mean()

Season
Fall      38.830085
Spring    39.293314
Summer    38.690885
Winter    38.366148
Name: Purchase Amount (USD), dtype: float64

Spring show the highest average purchase amount at $39.29 USD. However, seasonal differences are minimal, with average spending ranging from $38.37 to $39.29 USD.

In [188]:
# 4. Which states generate the highest sales volume?
df.groupby('Location')['Purchase Amount (USD)'].sum().sort_values(ascending=False).head(10)


Location
California       3474.548095
Nebraska         3195.731939
Delaware         3140.865589
West Virginia    3127.541031
Idaho            3097.347396
Indiana          3086.464246
Montana          3077.969533
Maryland         3048.546749
New York         3039.720209
Nevada           3032.907762
Name: Purchase Amount (USD), dtype: float64

The top three states with the highest sales volume are California, Nebraska, and Delaware, with sales total of $3474.55, $3195.73, and $3140.07 USD respectively. 

In [189]:
# 5. What are the most popular colors by season?
df.groupby(['Season'])['Color'].value_counts().groupby('Season').idxmax()

Season
Fall            (Fall, Brown)
Spring    (Spring, Baby blue)
Summer     (Summer, Lavender)
Winter        (Winter, Black)
Name: count, dtype: object

In Fall, the most popular color is Brown. In Spring, the most popular color is Baby Blue. In Summer, the most popular color is Lavender. In Winter, the most popular color is Black.

In [190]:
# 6. What is the most popular clothing item by season?
df.groupby(['Season'])['Item Purchased'].value_counts().groupby('Season').idxmax()

Season
Fall                (Fall, Socks)
Spring    (Spring, Running Shoes)
Summer           (Summer, Shorts)
Winter         (Winter, Leggings)
Name: count, dtype: object

In Fall, Spring, Summer, and Winter, the most popular clothing items are socks, running shoes, shorts, and leggings, respectively. Fall and Spring share a preference for footwear, while Summer and Winter show stronger popularity for lower‑body apparel.

In [191]:
# 7. What is the effect of promo codes on the dollar amount of purchases?
df.groupby('Promo Code Used')['Purchase Amount (USD)'].mean()

Promo Code Used
No     30.172085
Yes    50.023814
Name: Purchase Amount (USD), dtype: float64

Customers who use promo codes tend to spend more on average than those who do not. The difference in the average purchase amount between the two groups is $19.85 USD, indicating that promo‑code users make higher‑value purchases overall.

In [192]:
# 8. When do users leave a review?
df['Left Review'] = df['Review Rating'].notnull()
df.groupby(['Season', 'Left Review']).size().unstack()


Left Review,False,True
Season,,
Fall,669,387
Spring,376,205
Summer,602,340
Winter,822,499


The data shows that more customers chose not to submit a rating review than those who did. For example, in the Fall, 669 customers did not leave a review compared with 387 who did. Among those who did submit a review, Winter had the highest participation with 499 reviews, while Spring had the lowest with 205.